pageserver/src/layered_repository/timeline.rs

```diff
@@ -936,7 +936,7 @@ impl LayeredTimeline {
                         reconstruct_state,
                     )?;
                     cont_lsn = lsn_floor;
-                    traversal_path.push((result, cont_lsn, open_layer.clone()));
+                    traversal_path.push((result, cont_lsn, open_layer.filename()));
```

Comment:
Constructing the filename is probably slower than incrementing the reference count in the Arc. I'm not sure if that's measureable in practice, but `traversal_path` is only used in case of an error, and this is a fairly hot codepath, so let's try to keep it fast.